intake/catalog/tests/test_local.py

```diff
@@ -80,6 +80,17 @@ def test_use_source_plugin_from_config(catalog1):
     catalog1['use_example1'].get()
 
 
+def test_get_dir():
+    assert get_dir('s3://catalog.yml') == os.getcwd()
```

Comment:
This isn't right!
```
's3://mypath/afile' -> 's3://mypath/'
'https://example.com/catalog.yml' -> 'https://example.com/'
```

Otherwise relative paths in any remote catalog will fail - i.e., we should be able to move a directory+files with `{{ CATALOG_DIR }}` from local to remote and expect it to work. 
Glad we test for this.

Reply:
Great!  I'm glad we tested this too!

Do you really want the trailing `/`?

Reply:
Hm, interesting point. It shouldn't matter for most filesystems, but I suppose canonical would be no trailing `/`. Another thing that could/should be encoded into fsspec.

Reply:
I will make it so!  Also, opened an issue in (fsspec)[https://github.com/martindurant/filesystem_spec/issues/10]